insets-ui/src/main/java/com/google/accompanist/insets/ui/BottomNavigation.kt

```diff
@@ -47,19 +52,28 @@ fun BottomNavigation(
     backgroundColor: Color = MaterialTheme.colors.primarySurface,
     contentColor: Color = contentColorFor(backgroundColor),
     elevation: Dp = BottomNavigationDefaults.Elevation,
-    content: @Composable RowScope.() -> Unit
+    content: @Composable RowScope.() -> Unit,
 ) {
     Surface(
         color = backgroundColor,
+        contentColor = contentColor,
         elevation = elevation,
-        modifier = modifier
+        modifier = modifier,
     ) {
-        BottomNavigation(
-            backgroundColor = Color.Transparent,
-            contentColor = contentColor,
-            elevation = 0.dp,
-            modifier = Modifier.padding(contentPadding),
-            content = content
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .height(BottomNavigationHeight + contentPadding.calculateBottomPadding())
+                .padding(contentPadding)
```

Comment:
Tiny nit: If the `.padding` is applied before the `.height`, then you don't have to separately calculate the padding content:
```suggestion
                .padding(contentPadding)
                .height(BottomNavigationHeight)
```